### PR TITLE
:sparkles: Add login page

### PR DESCRIPTION
**Pull request by DonaCrio**
## What does this PR do?
It adds a login page

## Motivation (link to Trello card)
https://trello.com/c/wIqI2MWb

## Release Checklist

-   [ ] This code has automated tests
-   [ ] This code has documentation
-   [x] I have a plan to verify that these changes are working as expected after deploy

### CODE REVIEW

File: src/main/java/com/centralesupelec/chowchow/config/CorsFilter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  public void init(FilterConfig filterConfig) throws ServletException {}
+
+  @Override
+  public void doFilter(
+      ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+      throws IOException, ServletException {
+    HttpServletResponse response = (HttpServletResponse) servletResponse;
+    HttpServletRequest request = (HttpServletRequest) servletRequest;
+    String origin = request.getHeader("Origin");
+    response.setHeader("Access-Control-Allow-Origin", origin);
```
Comment (hinosxz) [inline src/main/java/com/centralesupelec/chowchow/config/CorsFilter.java]: Hacky!

File: src/main/java/com/centralesupelec/chowchow/config/CorsFilter.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CorsFilter implements Filter {
+
+  @Override
+  public void init(FilterConfig filterConfig) throws ServletException {}
+
+  @Override
+  public void doFilter(
```
Comment (hinosxz) [inline src/main/java/com/centralesupelec/chowchow/config/CorsFilter.java]: Let's try to refactor this in a follow-up ♻️

File: src/main/java/com/centralesupelec/chowchow/config/WebConfig.java [hunk 3]
```diff
@@ -11,6 +11,6 @@
 
   @Override
   public void addCorsMappings(CorsRegistry registry) {
-    registry.addMapping("/**");
+    registry.addMapping("/**").allowedOrigins("http://localhost:3000");
```
Comment (hinosxz) [inline src/main/java/com/centralesupelec/chowchow/config/WebConfig.java]: Do we need this?

File: src/main/javascript/src/components/Routes.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { RoutePath } from './RoutePath';
+import { PageLogin } from './authentication/PageLogin/PageLogin';
+import { PageLogout } from './authentication/PageLogout/PageLogout';
+import { AuthenticationState } from './authentication/types';
+
+export const Routes = () => {
+  const [authenticationState, setAuthenticationState] = React.useState<AuthenticationState>({
+    isAuthenticated: false,
+  });
+  console.log(authenticationState);
```
Comment (hinosxz) [inline src/main/javascript/src/components/Routes.tsx]: 😱

File: src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import {
+  Button, Form, Icon, Input,
+} from 'antd';
+import { FormComponentProps } from 'antd/es/form';
+
+import './login-form.scss';
+import { postLogin } from '../../../lib/api/login';
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx]: Avoid relative paths, please 🙂

File: src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import { AuthenticationState } from '../types';
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx]: Same here!

File: src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+const BLOCK = 'login-form';
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx]: I tried to use the BEM convention in the project
```suggestion
const BLOCK = 'authentication_login-form';
```

File: src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+type LoginFormProps = FormComponentProps<{username: string, password: string}>
+& {authenticationState: AuthenticationState, setAuthenticationState: (value: AuthenticationState) => void};
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx]: Is this prettified?

File: src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+      form.validateFields((err, values) => {
+        if (!err) {
+          const { username, password } = values;
+          postLogin(username, password)
+            .then(
+              res => {
+                if (res) {
+                  setAuthenticationState({ isAuthenticated: true });
+                }
+                console.log(authenticationState);
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx]: 😱

File: src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+          />,
+        )}
+      </Form.Item>
+      <Form.Item>
+        <Button type="primary" htmlType="submit" className={`${BLOCK}__button`}>
+          Log in
+        </Button>
+        Or
+        {' '}
+        <a href="/test" className={`${BLOCK}__register`}>Register</a>
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/LoginForm/LoginForm.tsx]: No a, please use ReactRouter `Link`

File: src/main/javascript/src/components/authentication/PageLogin/PageLogin.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { Card } from 'antd';
+
+import { LoginForm } from '../LoginForm/LoginForm';
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/PageLogin/PageLogin.tsx]: Import absolute

File: src/main/javascript/src/components/authentication/PageLogin/PageLogin.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+import './page-login.scss';
+import { AuthenticationState } from '../types';
+
+const BLOCK = 'page-login';
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/PageLogin/PageLogin.tsx]:
```suggestion
const BLOCK = 'authentication_page-login';
```

File: src/main/javascript/src/components/authentication/PageLogout/PageLogout.tsx [hunk 13]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import { Card } from 'antd';
+
+import { AuthenticationState } from '../types';
+import { postLogout } from '../../../lib/api/login';
```
Comment (hinosxz) [inline src/main/javascript/src/components/authentication/PageLogout/PageLogout.tsx]: Import absolute

File: src/main/javascript/src/components/home/PageHome/PageHome.tsx [hunk 14]
```diff
@@ -12,6 +12,7 @@ import { PageShow } from 'components/show/PageShow/PageShow';
 import { ShowView } from 'components/home/ShowView/ShowView';
 
 import './page-home.scss';
+import { RoutePath } from '../../RoutePath';
```
Comment (hinosxz) [inline src/main/javascript/src/components/home/PageHome/PageHome.tsx]: Import absolute

File: src/main/javascript/src/index.tsx [hunk 15]
```diff
@@ -1,9 +1,7 @@
...
 import ReactDOM from 'react-dom';
-import { BrowserRouter as Router } from 'react-router-dom';
-
-import { PageHome } from './components/home/PageHome/PageHome';
 
 import './index.scss';
+import { Routes } from './components/Routes';
 
-ReactDOM.render(<Router><PageHome /></Router>, document.getElementById('root'));
+ReactDOM.render(<Routes />, document.getElementById('root'));
```
Comment (hinosxz) [inline src/main/javascript/src/index.tsx]: 👏

File: src/main/java/com/centralesupelec/chowchow/config/CorsFilter.java (hunk 1 above)
Comment (DonaCrio) [reply src/main/java/com/centralesupelec/chowchow/config/CorsFilter.java]: https://trello.com/c/9FZTcFTX/49-add-env-variable-for-cors-origin

**Pull request merged**